### DISCUSSION TITLE
Question on the breaking change in v4

### DISCUSSION THREAD

**Discussion by pcb1962** [YorkshireIoT/ha-google-fit#189] [General]
Hi, if this version creates a new device will it also create new entities - so existing graphs of say sensor.weight will not update further and a new entity will need to be added for future values?
If this is so, and I rename the new entity same as the old will the existing history be retained and updated going forwrd?
Thanks

**Comment by YorkshireIoT** [comment]
I've done the upgrade on my own HA and all the entities stayed the same, so it should just be the same for you.

I have to admit that I did expect HA to create new entities, hence the breaking change notice, so was pleasantly surprised that everything worked normally.
